icicle/appUtils/msm/msm.cu

```diff
@@ -323,125 +324,125 @@ void batched_bucket_method_msm(unsigned bitsize, unsigned c, S *scalars, A *poin
   unsigned bm_bitsize = ceil(log2(nof_bms));
   unsigned nof_buckets = (nof_bms<<c);
   unsigned total_nof_buckets = nof_buckets*batch_size;
-  cudaMalloc(&buckets, sizeof(P) * total_nof_buckets); 
+  cudaMallocAsync(&buckets, sizeof(P) * total_nof_buckets, stream); 
 
   //lanch the bucket initialization kernel with maximum threads
   unsigned NUM_THREADS = 1 << 10;
   unsigned NUM_BLOCKS = (total_nof_buckets + NUM_THREADS - 1) / NUM_THREADS;
-  initialize_buckets_kernel<<<NUM_BLOCKS, NUM_THREADS>>>(buckets, total_nof_buckets); 
+  initialize_buckets_kernel<<<NUM_BLOCKS, NUM_THREADS, 0, stream>>>(buckets, total_nof_buckets); 
 
   unsigned *bucket_indices;
   unsigned *point_indices;
-  cudaMalloc(&bucket_indices, sizeof(unsigned) * (total_size * nof_bms + msm_size));
-  cudaMalloc(&point_indices, sizeof(unsigned) * (total_size * nof_bms + msm_size));
+  cudaMallocAsync(&bucket_indices, sizeof(unsigned) * (total_size * nof_bms + msm_size), stream);
+  cudaMallocAsync(&point_indices, sizeof(unsigned) * (total_size * nof_bms + msm_size), stream);
 
   //split scalars into digits
   NUM_THREADS = 1 << 8;
   NUM_BLOCKS = (total_size * nof_bms + msm_size + NUM_THREADS - 1) / NUM_THREADS;
-  split_scalars_kernel<<<NUM_BLOCKS, NUM_THREADS>>>(bucket_indices + msm_size, point_indices + msm_size, d_scalars, total_size, 
+  split_scalars_kernel<<<NUM_BLOCKS, NUM_THREADS, 0, stream>>>(bucket_indices + msm_size, point_indices + msm_size, d_scalars, total_size, 
                                                     msm_log_size, nof_bms, bm_bitsize, c); //+size - leaving the first bm free for the out of place sort later
 
   //sort indices - the indices are sorted from smallest to largest in order to group together the points that belong to each bucket
   unsigned *sorted_bucket_indices;
   unsigned *sorted_point_indices;
-  cudaMalloc(&sorted_bucket_indices, sizeof(unsigned) * (total_size * nof_bms));
-  cudaMalloc(&sorted_point_indices, sizeof(unsigned) * (total_size * nof_bms));
+  cudaMallocAsync(&sorted_bucket_indices, sizeof(unsigned) * (total_size * nof_bms), stream);
+  cudaMallocAsync(&sorted_point_indices, sizeof(unsigned) * (total_size * nof_bms), stream);
 
   unsigned *sort_indices_temp_storage{};
   size_t sort_indices_temp_storage_bytes;
   cub::DeviceRadixSort::SortPairs(sort_indices_temp_storage, sort_indices_temp_storage_bytes, bucket_indices + msm_size, sorted_bucket_indices,
-                                 point_indices + msm_size, sorted_point_indices, total_size * nof_bms);
-  cudaMalloc(&sort_indices_temp_storage, sort_indices_temp_storage_bytes);
+                                 point_indices + msm_size, sorted_point_indices, total_size * nof_bms, 0, sizeof(unsigned)*8, stream);
```

Comment:
why 8?

Reply:
ok, so maybe you can just add a comment in the code explaining

Reply:
Added comment in both `batched_bucket_method_msm` and `bucket_method_msm`